morf-core/src/main/java/org/alfasoftware/morf/upgrade/UpgradeScriptAdditionsProvider.java

```diff
@@ -0,0 +1,57 @@
+package org.alfasoftware.morf.upgrade;
+
+
+import com.google.inject.ImplementedBy;
+import com.google.inject.Inject;
+import org.alfasoftware.morf.upgrade.additions.UpgradeScriptAddition;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.function.Predicate;
+
+@ImplementedBy(UpgradeScriptAdditionsProvider.DefaultScriptAdditions.class)
+public interface UpgradeScriptAdditionsProvider {
+
+
+    /**
+     * Returns all script additions with the filtering criteria applied.
+     * The filtering logic should be provided by calling {@link #setAllowedPredicate(Predicate)} first.
+     * @return
+     */
+    default Set<UpgradeScriptAddition> getUpgradeScriptAdditions() {
+        return Collections.emptySet();
+    }
+
+
+    /**
+     * Allows for filtering of script additions.
+     * @param scriptAdditionsPredicate
+     */
+    default void setAllowedPredicate(Predicate<UpgradeScriptAddition> scriptAdditionsPredicate) {
+    }

```

Comment:
Just wondering: Perhaps injecting UpgradeScriptAdditionsProviderFactory, which would get the correct Predicate would be better than having a setter, which you need to call from all relevant places, risking some places will be forgotten.